### PR TITLE
Add test for python 3.11

### LINKED ISSUE #524: [FEATURE REQUEST] Support for Python 3.11
**Is your feature request related to a problem? Please describe.**

I am always frustrated when a new version of Python ships but my favorite tools are yet not compatible with it.
* https://devguide.python.org/versions
* https://www.python.org/download/pre-releases
* PyCQA/pylint#5920

**Describe the solution you'd like**

[Python 3.11 release candidate 2](https://www.python.org/download/pre-releases) is tested and passes on this repo's continuous integration system.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Description

Closes #524

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.7', '3.8', '3.9', '3.10']
+        python-version: ['3.7', '3.8', '3.9', '3.10', "3.11"]
```
Comment (cclauss) [inline .github/workflows/tests.yml:9]:
```suggestion
        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
```
Or double quotes everywhere.

Comment (Pierre-Sassoulas) [reply .github/workflows/tests.yml:9]: We can apply prettier on yml now that there won't be conflict as we merged #536

Comment (Pierre-Sassoulas) [reply .github/workflows/tests.yml:9]: https://github.com/PyCQA/prospector/pull/541

**Comment by Pierre-Sassoulas** [comment]
Closing in favor of #543

**Pull request closed without merging**